internal/p2p/router.go

```diff
@@ -591,9 +584,8 @@ LOOP:
 		switch {
 		case errors.Is(err, context.Canceled):
 			break LOOP
-		case err != nil:
-			r.logger.Error("failed to find next peer to dial", "err", err)
-			break LOOP
+		case address == NodeAddress{}:
+			continue LOOP
```

Comment:
This only happens when the context is cancelled right?

Reply:
i.e. `DialNext` either always returns a node address or returns and empty one and `ctx.Err()`

Reply:
yeah, that's the case, but even if it weren't we shouldn't abort dialing peers just because we had a problem getting an error from dialnext: until the context is canceled we should keep  trying to dial.

Reply:
I see, I still think it's reasonable to have the code be a little defensive here.